### DISCUSSION TITLE
API: Not sure how to set samplers after recent update....?

### DISCUSSION THREAD

**Discussion by anothertal3** [vladmandic/sdnext#2323] [Q&A] [answered]
Until yesterday I've used the API to send tasks to SD.NEXT using, for example, the following command:
![image](https://github.com/vladmandic/automatic/assets/8721770/c1643a3c-e324-4227-b088-dbefe48c3437)

However, I haven't been able to process anything since the recent update. The main reason appears to be that the sampler parameter will not work anymore:
> 01:18:36-284348 ERROR    Attempting to use unknown sampler: DPM++ 2M Karras

Within the UI you are now required to a) set "DPM++ 2M" and b) set "Karras" separately if I understand it correctly. How would I achieve the same using the API? I don't see any related "sigma algorithm" parameter?

Thank you for your help

**Comment by vladmandic** [comment, marked as answer]
Karras, sigma, etc - those are all settings, not params, so you set them as you would settings. And then call txt2img api with sampler name as usual at it will use whatever setting you applied.

**Reply by anothertal3** [reply]
First of all: your response time is insane! Thank you very much.

Second: understood. I will experiment in that regard. If I recall correctly there might even be an API method to change "settings" so this might only be a matter of sending two requests then.

**Reply by vladmandic** [reply]
yes, settings are set using single api call and its near-instant, can be 1 or 100 settings as payload for that call.
